core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -170,12 +259,26 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalString(message);
         out.writeThrowable(failure);
         out.writeVInt(failedAllocations);
+        lastAllocationStatus.writeTo(out);
     }
 
     public UnassignedInfo readFrom(StreamInput in) throws IOException {
         return new UnassignedInfo(in);
     }
 
+    public UnassignedInfo withLastAllocationStatus(AllocationStatus status) {
```

Comment:
This method is only used in one place. We can just call the constructor there if its needed.